python-client/giskard/scanner/correlation/spurious_correlation_detector.py

```diff
@@ -93,83 +113,11 @@ def _get_measure_fn(self):
         raise ValueError(f"Unknown method `{self.method}`")
 
 
-@dataclass
-class SpuriousCorrelationInfo:
-    feature: str
-    slice_fn: SlicingFunction
-    metric_value: float
-    metric_name: str
-    threshold: float
-    predictions: pd.DataFrame
-
-
-class SpuriousCorrelationIssue(Issue):
-    group = "Spurious correlation"
-
-    @property
-    def features(self):
-        return [self.info.feature]
-
-    @property
-    def domain(self) -> str:
-        return str(self.info.slice_fn)
-
-    @property
-    def metric(self) -> str:
-        return f"Nominal association ({self.info.metric_name})"
-
-    @property
-    def deviation(self) -> str:
-        plabel, p = self.info.predictions.index[0], self.info.predictions.iloc[0]
-
-        return f"Prediction {self.dataset.target} = `{plabel}` for {p * 100:.2f}% of samples in the slice"
-
-    @property
-    def slicing_fn(self):
-        return self.info.slice_fn
-
-    @property
-    def description(self) -> str:
-        pred = self.model.predict(self.dataset.slice(self.info.slice_fn)).prediction
-        classes = pd.Series(pred).value_counts(normalize=True)
-        plabel, p = classes.index[0], classes.iloc[0]
-        return f"Data slice {self.info.slice_fn} seems to be highly associated to prediction {self.dataset.target} = `{plabel}` ({p * 100:.2f}% of predictions in the data slice)."
-
-    # @lru_cache
-    def examples(self, n=3):
-        extractor = ExampleExtractor(self)
-        return extractor.get_examples_dataframe(n, with_prediction=1)
-
-    @property
-    def importance(self) -> float:
-        return self.info.metric_value
-
-    def generate_tests(self, with_names=False) -> list:
-        test_fn = _metric_to_test_object(self.info.metric_name)
-
-        if test_fn is None:
-            return []
-
-        tests = [
-            test_fn(
-                model=self.model,
-                dataset=self.dataset,
-                slicing_function=self.info.slice_fn,
-                threshold=self.info.threshold,
-            )
-        ]
-
-        if with_names:
-            names = [f"{self.info.metric_name} on data slice “{self.info.slice_fn}”"]
-            return list(zip(tests, names))
-
-        return tests
-
-
 _metric_test_mapping = {
-    "Cramer's V": "test_cramer_v",
-    "Mutual information": "test_mutual_information",
-    "Theil's U": "test_theil_u",
+    "cramer": "test_cramer_v",
```

Comment:
in places like this can you reference a test function itself, not just it's name? It'll be difficult to remember to modify these strings in case we do automatic test method renaming

Reply:
I think it was done like this to only import the tests lazily (this will only be called on `generate_test_suite`). I kept this in the current refactoring, tests are initialized lazily. (We wanted to minimize the big imports of testing as far as I understood.)

In principle these functions are part of our public API (`testing`), so I exclude the option of a light-hearted renaming.